lib/apiv2/felixconfig.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiv2
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	KindFelixConfiguration     = "FelixConfiguration"
+	KindFelixConfigurationList = "FelixConfigurationList"
+)
+
+// Felix Configuration contains the configuration for Felix.
+type FelixConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the FelixConfiguration.
+	Spec FelixConfigurationSpec `json:"spec,omitempty"`
+}
+
+// TODO: Add validation on LogSeverityScreen, LogSeveritySys, LogSeverityFile, DatastoreType,
+// DefaultEndpointToHostAction, IptablesFilterAllowAction, IptablesMangleAllowAction,
+// ChainInsertModefor valid values.
+// FelixConfigurationSpec contains the values of the Felix configuration.
+type FelixConfigurationSpec struct {
+	UseInternalDataplaneDriver *bool  `json:"useInternalDataplaneDriver,omitempty"`
+	DataplaneDriver            string `json:"dataplaneDriver,omitempty"`
+
+	DatastoreType string `json:"datastoreType,omitempty"`
+
+	TyphaAddr           string         `json:"typhaAddr,omitempty"`
+	TyphaK8sServiceName string         `json:"typhaK8sServiceName,omitempty"`
+	TyphaK8sNamespace   string         `json:"typhaK8sNamespace,omitempty"`
+	TyphaReadTimeout    *time.Duration `json:"typhaReadTimeout"`
+	TyphaWriteTimeout   *time.Duration `json:"typhaWriteTimeout"`
+
+	Ipv6Support    *bool `json:"ipv6Support,omitempty"`
+	IgnoreLooseRPF *bool `json:"ignoreLooseRPF,omitempty"`
+
+	RouteRefreshInterval               *time.Duration `json:"routeRefreshInterval,omitempty"`
```

Comment:
Sorry to expand the scope here, but could you do the following for all of the duration/interval fields:
-  Convert from *time.Duration to *int
-  Make sure each field has the correct Suffix (which I think will be "Secs", but could you cross reference with: https://docs.projectcalico.org/v2.6/reference/felix/configuration to make sure).
-  For every fields where you are modifying the field name from the original, could you add a configname tag that specifies the original field name.   e.g

```
	RouteRefreshIntervalSecs           *int `json:"routeRefreshInterval,omitempty" configname:"RouteRefreshInterval"`
```